### ISSUE TITLE
`install_git` removes the "@" from the url

### ISSUE THREAD

**Issue by stla** [r-lib/remotes#646]
Hello,

 I'm using Windows 10 and I'm trying to install a package from Azure DevOps.

```
> remotes::install_git("https://stephanexlaurent@dev.azure.com/stephanexlaurent/MyProject/_git/MyPackage", git="external")
Error: Failed to install 'unknown package' from Git:
  Command failed (128)
In addition: Warning messages:
1: In system(full, intern = TRUE, ignore.stderr = quiet) :
  running command '"C:\PORTAB~1\cmder\vendor\GIT-FO~1\cmd\git.exe" ls-remote https://stephanexlaurent dev.azure.com/stephanexlaurent/MyProject/_git/MyPackage' had status 128
```

Observe the warning message in `system`: the "@" has been replaced with a white space.

**Comment by achimgaedke** [comment]
Take the regexp from https://github.com/r-lib/remotes/blame/37f5c96f442a8b7696e024dd6badfa83048067fd/R/install-git.R#L82 and try it out:

```
meta <- rematch2::re_match("ssh://git@git.very-important.com:22/analytics/use-this.git",
                           "(?<url>(?:git@)?[^@]*)(?:@(?<ref>.*))?")
```


`meta$url` is `'ssh://git'` and the rest is in `meta$ref` 😿

**Comment by vspinu** [comment]
Works with v2.3.0:

```R
> url <- "ssh://git@git.host.com:7999/proj/name.git"
> remotes:::git_remote(url)
$url
[1] "ssh://git@git.host.com:7999/proj/name.git"

$subdir
NULL

$ref
NULL

$credentials
NULL

attr(,"class")
[1] "git2r_remote" "remote"      
```


Doesn't work in dev nor the released CRAN version v2.4.0:

```R
> url <- "ssh://git@git.host.com:7999/proj/name.git"
> remotes:::git_remote(url)
$url
[1] "ssh://git"

$subdir
NULL

$ref
[1] "git.host.com:7999/proj/name.git"

$credentials
NULL

attr(,"class")
[1] "git2r_remote" "remote"      
```

Thus `devtools::install_git` on all git urls currently fails.

**Comment by ThomUK** [comment]
This issue was fixed in #658, and should be closed.
